Output gateway VPC endpoint prefix list IDs

mattburgess opened the pull request:
These may need to be referenced by a caller's security group rules

Commit:
Output gateway VPC endpoint prefix list IDs

These may need to be referenced by a caller's security group rules

dan-hill2802 reviewed (changes requested):
Syntax is wrong, replace `.0.` with `[0]`

mattburgess commented:
No, that syntax is right, and tested locally.  It's not exactly intuitive, and I couldn't find it in the docs though.  https://www.terraform.io/docs/configuration-0-11/interpolation.html#user-list-variables shows your syntax but that's when you want to return an entire item from a list. If you're trying to reference an attribute of a list item then my syntax is the only one that works:

This is what I get with `aws_vpc_endpoint.s3[0].prefix_list_id`:

`Error downloading modules: Error loading modules: module vpc: Error loading .terraform/modules/784d75d77d86b31f5057440df3fdb888/outputs.tf: Error reading config for output s3_prefix_list_id: parse error at 1:43: expected : but found "."`

I notice, though that the outputs just below these ones could do with a similar condition on them so that callers don't need to do the element indexing themselves. Are you OK for me to push that in this PR or do you want a separate one?

mattburgess commented:
Interestingly, this also works, but I think I prefer my syntax: `aws_vpc_endpoint.s3.*.prefix_list_id[0]`

dan-hill2802 commented:
How strange. Yeah you can just alter the other outputs as part of this PR

Commit:
Output string rather than list for SSM IAM outputs

This means that callers don't have to index into the list to obtain the
single output they need to use.
